app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/cloud/AppEngineDeploy.java

```diff
@@ -147,9 +148,14 @@ public void deploy(
             .build();
 
     // show a warning notification if the cloud sdk version is not supported
-    CloudSdkVersionNotifier.getInstance().notifyIfUnsupportedVersion();
+    CloudSdkVersionNotifier.getInstance().notifyIfVersionOutOfDate();
+    CloudSdkVersionNotifier.getInstance().notifyIfVersionParseError();
 
-    helper.createGcloud(loggingHandler).newDeployment(processHandler).deploy(configuration);
+    try {
+      helper.createGcloud(loggingHandler).newDeployment(processHandler).deploy(configuration);
+    } catch (CloudSdkOutOfDateException ex) {
```

Comment:
So being out of date doesn't in fact mean Cloud SDK cannot successfully deploy application, and the new flow is more user friendly?

Reply:
Out of date always threw an exception. Previously this exception was not caught and the IDE would explode (after showing the nice notification to the user). Now I'm catching the exception to prevent blowing up.

In both cases though, appengine-plugins-core would not let the operation proceed when out of date.

Does that answer the question?